### PR TITLE
Products: move conditional titles to plans list

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to #

## Proposed Changes

In the plans comparison grid, there is a use case for showing different labels/text for features based on the plan. Example:
image

This was achieved using the `getConditionalTitle` function in `packages/calypso-products/src/features-list.tsx`. This function was used to output different descriptions for the feature based on the `planSlug` passed to the function. Additionally, the plan definition in `packages/calypso-products/src/plans-list.tsx` also contained a `get2023PlanComparisonConditionalFeatures` function, which contained a list of features for that plan that have conditional titles (fetched using `getConditionalTitle`).

The approach caused issues with tree-shaking (p7H4VZ-4S4-p2). Conceptually, it also meant that plan-specific logic had leaked into the features list.

This PR removes the `getConditionalTitle` function from `features-list.tsx` and the `get2023PlanComparisonConditionalFeatures` from `plans-list.tsx`. This has been now replaced by a single function: `get2023PlanComparisonFeatureLabels`. (Eventually, we'll need to refactor out all the `2023` references 😅 )

The `get2023PlanComparisonFeatureLabels` is a map of the feature slug to the text for that plan. In the future, it also makes it easier to add plan-specific strings without touching the features list.

## Testing Instructions

* Confirm that the comparison grid is the same as production for the following scenarios:
  * `/start/plans`
  * Trail map variants:
    * `/start/plans?flags=onboarding/trail-map-feature-grid-structure`
    * `/start/plans?flags=onboarding/trail-map-feature-grid-copy`
    * `/start/plans?flags=onboarding/trail-map-feature-grid`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/calypso-products/src/features-list.tsx [hunk 1]
```diff
@@ -554,36 +546,14 @@ const FEATURES_LIST: FeatureList = {
...
-			return '—';
-		},
 	},
 
 	[ WPCOM_FEATURES_PREMIUM_THEMES_UNLIMITED ]: {
 		getSlug: () => WPCOM_FEATURES_PREMIUM_THEMES_UNLIMITED,
-		getTitle: ( { planSlug = undefined } = {} ) => {
-			if ( planSlug && isWooExpressPlan( planSlug ) ) {
-				return i18n.translate( 'Beautiful themes' );
-			}
```
Comment (aneeshd16) [inline packages/calypso-products/src/features-list.tsx:570]: WooExpress has been retired, so this is dead code.

File: packages/calypso-products/src/features-list.tsx [hunk 2]
```diff
@@ -554,36 +546,14 @@ const FEATURES_LIST: FeatureList = {
...
+		getTitle: () => {
 			return i18n.translate( 'Access to all premium themes' );
 		},
-		getDescription: ( { planSlug = undefined } = {} ) => {
-			if ( planSlug && isWooExpressPlan( planSlug ) ) {
-				return i18n.translate( 'Switch between a collection of beautiful themes.' );
-			}
```
Comment (aneeshd16) [inline packages/calypso-products/src/features-list.tsx:576]: Same as above.

**Review by southp — APPROVED** [review]
Works as described. It's a fascinating read to see the unexpected circular dependency between the two modules has such an impact :) By cleaning it up here, I also found it easier to reason now.

**Comment by aneeshd16** [comment]
Thanks @southp, for the review! I am working on something else and I can take up the rebase later this week. I also want to rename `get2023PlanComparisonFeatureLabels` -> `getPlanComparisonFeatureLabels` as adding a new function with `2023` does not make sense to me anymore. 🙂

**Comment by southp** [comment]
> I also want to rename get2023PlanComparisonFeatureLabels -> getPlanComparisonFeatureLabels as adding a new function with 2023 does not make sense to me anymore

That sounds fantastic. We should definitely move on from 2023 😆

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14541149

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14541149).

**Pull request merged**